### PR TITLE
Refactoring and clean up of the CORSIKA code

### PR DESCRIPTION
**Pull request by soleti**
Following @gaponenko suggestions I implemented a new way to read the CORSIKA binary files. Now there is no need of specifying the format of the input binary file ("compact" or "full"): a common method is able to distinguish between the two and read the information accordingly. The run number is now obtained from the binary file content and not from the filename. Also, the "lowE" and "highE" parameters that were specified in the fcl can be read from the binary file, so they have been removed from the fcl configuration.

### CODE REVIEW

File: Sources/inc/CosmicCORSIKA.hh [hunk 1]
```diff
@@ -229,30 +234,35 @@ namespace mu2e {
       static constexpr float _GeV2MeV = CLHEP::GeV / CLHEP::MeV;
       static constexpr float _cm2mm = CLHEP::cm / CLHEP::mm;
       static constexpr float _ns2s = CLHEP::ns / CLHEP::s;
+      static constexpr unsigned _fbsize_words = 5733 + 2;
```
Comment (gaponenko) [inline Sources/inc/CosmicCORSIKA.hh:237]: A comment about 5733 + 2 would be helpful.

File: Sources/inc/CosmicCORSIKA.hh [hunk 2]
```diff
@@ -229,30 +234,35 @@ namespace mu2e {
...
 
       CLHEP::Hep3Vector _cosmicReferencePointInMu2e;
       float _fluxConstant = 1.8e4;
```
Comment (gaponenko) [inline Sources/inc/CosmicCORSIKA.hh:240]: I  wanted to know why 1.8e4, but then realized that the number is always owerwritten by the corresponding setting in the fcl file.   May be just get rid of the number here?

File: Sources/src/CosmicCORSIKA.cc [hunk 3]
```diff
@@ -23,19 +23,74 @@ namespace mu2e {
...
+        throw std::runtime_error("Error: record size not a multiple of 4");
+      }
+
+      // We will be looking at at least 8 bytes to determine the
+      // input file format, and all real CORSIKA records are longer
+      // than that.
+      if(reclen < 2*4) {
+        throw std::runtime_error("Error: reclen too small");
+      }
+
```
Comment (gaponenko) [inline Sources/src/CosmicCORSIKA.cc:53]: I suggest to also add a protection agains reclen > 4*_fbsize_words,  so we do not overwrite memory on corrupt inputs.

File: Sources/inc/CosmicCORSIKA.hh (hunk 2 above)
Comment (gaponenko) [reply Sources/inc/CosmicCORSIKA.hh:240]: That comment went onto a wrong line.  It was about the _fluxConstant.

**Comment by brownd1978** [comment]
@FNALbuild run build test

File: Sources/inc/CosmicCORSIKA.hh (hunk 1 above)
Comment (soleti) [reply Sources/inc/CosmicCORSIKA.hh:237]: Added, thanks.

File: Sources/inc/CosmicCORSIKA.hh (hunk 2 above)
Comment (soleti) [reply Sources/inc/CosmicCORSIKA.hh:240]: 1.8e4 is the intensity of cosmic rays primary nucleons as quoted by the PDG and it's needed for the live-time calculation. I would still keep it in the code for clarity. I added a comment.

File: Sources/src/CosmicCORSIKA.cc (hunk 3 above)
Comment (soleti) [reply Sources/src/CosmicCORSIKA.cc:53]: Added, thanks.

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Pull request merged**